### PR TITLE
docs: describe compute resources config

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Updates the _Deploying_ guide
Adds a section to _Configuring a deployment_  that provides information on configuring cpu and memory resources, and why it might be required.  Links to the [resources](https://strimzi.io/docs/operators/in-development/configuring.html#con-common-configuration-resources-reference) section in the _Configuring_ guide for more information.

Plus, few small edits from review.

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by PaulRMellor** [comment]
This PR fixes #977

File: documentation/modules/deploying/con-deploy-options-scope.adoc [hunk 1]
```diff
@@ -31,3 +31,16 @@ Strimzi supports additional deployment options to monitor your deployment.
...
+By default, the Strimzi deployment does not specify requests and limits for CPU and memory resources.
+It's important for Kafka to have enough of these resources to run properly, but not more than they need if they are sharing resources.
+If you don't set limits, a container could use all the available memory.
+
+The amount of resources to reserve depends on the specific requirements of the deployment.
+The number of producers and consumers has an impact, as does the size of messages.
+You might monitor data throughput and other metrics to get an idea of the resources needed.
+
+You can set resource requests and limits for each container in a pod.
+If you want to set requests and limits, you link:{BookURLUsing}#con-common-configuration-resources-reference[add resources configuration to Strimzi custom resources].
```
Comment (scholzj) [inline documentation/modules/deploying/con-deploy-options-scope.adoc:46]: What about something like this?

```suggestion
== CPU and memory resource limits and requests

By default, the Strimzi Cluster Operator does not specify requests and limits for CPU and memory resources for any operands it deploys.
Having sufficient resources is important for applications like Kafka to be stable and deliver good performance.
The right amount of resources you should use depends on the specific requirements and use-cases.
You should consider configuring the CPU and memory resources.
You can set resource requests and limits for each container in the link:{BookURLUsing}#con-common-configuration-resources-reference[Strimzi custom resources].
```

The middle part from your test would apply well for Kafka brokers. But not for the other components. It is quite hard to strike the right level of encouragement without making things scary.

File: documentation/modules/deploying/con-deploy-options-scope.adoc [hunk 2]
```diff
@@ -31,3 +31,16 @@ Strimzi supports additional deployment options to monitor your deployment.
 * Extract metrics and monitor Kafka components by xref:assembly-metrics-setup-str[deploying Prometheus and Grafana with your Kafka cluster].
 * Extract additional metrics, particularly related to monitoring consumer lag, by xref:proc-metrics-kafka-deploy-options-{context}[deploying Kafka Exporter with your Kafka cluster].
 * Track messages end-to-end by link:{BookURLUsing}#assembly-distributed-tracing-str[setting up distributed tracing^].
+
+== CPU and memory resource limits and requests
+
+By default, the Strimzi deployment does not specify requests and limits for CPU and memory resources.
+It's important for Kafka to have enough of these resources to run properly, but not more than they need if they are sharing resources.
+If you don't set limits, a container could use all the available memory.
```
Comment (joaedwar) [inline documentation/modules/deploying/con-deploy-options-scope.adoc:39]: I know IBM says contractions are ok now, but I still think minimizing use of contractions is better for translation purposes.

File: documentation/modules/deploying/con-deploy-options-scope.adoc [hunk 3]
```diff
@@ -31,3 +31,16 @@ Strimzi supports additional deployment options to monitor your deployment.
...
+
+The amount of resources to reserve depends on the specific requirements of the deployment.
+The number of producers and consumers has an impact, as does the size of messages.
```
Comment (joaedwar) [inline documentation/modules/deploying/con-deploy-options-scope.adoc:42]: Please specify has an "impact" on what

File: documentation/modules/deploying/con-deploy-prereqs.adoc [hunk 4]
```diff
@@ -5,10 +5,10 @@
 [id='deploy-prereqs-{context}']
 = Deployment prerequisites
 
-To deploy Strimzi, make sure that:
+To deploy Strimzi, you'll need the following:
```
Comment (joaedwar) [inline documentation/modules/deploying/con-deploy-prereqs.adoc:8]: Please refrain from using you'll. I think that is a particularly difficult one for translation.

**Review by joaedwar — APPROVED** [review]
lgtm. Just a few minor suggestions. Thanks a million

File: documentation/modules/deploying/con-deploy-options-scope.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-options-scope.adoc:46]: Reduced version is good :+1:

File: documentation/modules/deploying/con-deploy-options-scope.adoc [hunk 5]
```diff
@@ -31,3 +31,16 @@ Strimzi supports additional deployment options to monitor your deployment.
 * Extract metrics and monitor Kafka components by xref:assembly-metrics-setup-str[deploying Prometheus and Grafana with your Kafka cluster].
 * Extract additional metrics, particularly related to monitoring consumer lag, by xref:proc-metrics-kafka-deploy-options-{context}[deploying Kafka Exporter with your Kafka cluster].
 * Track messages end-to-end by link:{BookURLUsing}#assembly-distributed-tracing-str[setting up distributed tracing^].
+
+== CPU and memory resource limits and requests
+
+By default, the Strimzi deployment does not specify requests and limits for CPU and memory resources.
+It's important for Kafka to have enough of these resources to run properly, but not more than they need if they are sharing resources.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-options-scope.adoc:38]: Updated

File: documentation/modules/deploying/con-deploy-options-scope.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-options-scope.adoc:39]: updated

File: documentation/modules/deploying/con-deploy-options-scope.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-options-scope.adoc:42]: updated

**Review by scholzj — APPROVED** [review]

**Pull request merged**